### ISSUE TITLE
Compiler gets stuck in Infinite (so far) loop

### ISSUE THREAD

**Issue by Validark** [ziglang/zig#19955]
### Zig Version

0.13.0-dev.75+5c9eb4081

### Steps to Reproduce and Observed Behavior

https://zig.godbolt.org/z/GYdeh839e

Compiling this code for Zen 4:

```zig
export fn popCount(x: @Vector(8, u64)) u64 {
    return @reduce(.Add, @popCount(x));
}
```

Takes forever (so far). I have reproduced this on my machine for the version indicated (`0.13.0-dev.75+5c9eb4081`), and also noted that it times out 0.11.0 and 0.12.0 in compiler explorer.

### Expected Behavior

The program to terminate at some point, preferably compiling the code!

**Comment by xdBronch** [comment]
this isnt reproducing for me locally on `0.13.0-dev.201+86d8688c7`. fixed in the llvm 18 upgrade maybe?

**Comment by mikdusan** [comment]
that's a dupe: andrew filed https://github.com/llvm/llvm-project/issues/90356 and fixed in 18.1.x

confirmed:
- `zig 0.12.0` w/ `llvm-17.0.6 release+assert` trips assertion
- `zig 0.12.0` w/ `llvm-17.0.6 release` spins infinite
- `zig master` w/ `llvm-18.1.5 release+assert` does not trip assertion
- `zig master` w/ `llvm-18.1.5 release` builds object

**Comment by mlugg** [comment]
Cool!
